Add flag to turn off auto exit with pm2-docker

dguo opened the pull request:
Because we want auto exit to be on by default for pm2-docker, the flag needs to be changed to explicitly turn it off. See the example at https://tj.github.io/commander.js/#Command.prototype.option for a "simple boolean defaulting to true".

The suggested fix was:
```js
if (commander.autoExit !== false) {
 autoExit();
}
```
But I couldn't find a way to actually pass "false" as a value from the CLI.

If you would like to add a test for this change, l could use some help on that.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no (print a warning message if --auto-exit is provided though)
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #3206 
| License       | MIT
| Doc PR        | https://github.com/pm2-hive/pm2-hive.github.io/pull/116

dguo commented on lib/binaries/Runtime4Docker.js:
```diff
@@ -24,12 +24,12 @@ commander.version(pkg.version)
   .option('--watch', 'watch and restart application on file change')
   .option('--error <path>', 'error log file destination (default disabled)', '/dev/null')
   .option('--output <path>', 'output log file destination (default disabled)', '/dev/null')
-  .usage('pm2-docker app.js');
```
The help usage was `Usage: pm2-docker pm2-docker app.js`, so this change removes the duplicate `pm2-docker`.

vmarchaud commented:
I think we can let a fake flag to maintain backward compatibility (cause a lot of people might using this flag), and we could even add a error message when the flag is used, something like : 
 `Warning: --auto-exit has been removed, it's now the default behavior, if you want to disable it a new flag has been introduced : --no-auto-exit.`) 
What do you think ?

dguo commented:
That makes sense. I added the warning message and rebased. This should be a backwards compatible change now.

Instead of leaving the `--auto-exit` flag in, I added a check to `process.argv`. I tried the fake flag approach, but Commander doesn't seem to handle having both `--foo` and `--no-foo` options. The value would come back as `false`, even when the flag was provided. 

Another approach would be to use Commander's [allowUnknownOption](https://tj.github.io/commander.js/#Command.prototype.allowUnknownOption), but then we would still have to check `process.argv` to provide the warning message, and we would lose the error handling for truly unknown options.

vmarchaud commented:
That sounds good good to me, thanks for taking the time to make this PR :)
